subprojects/plugins/src/integTest/groovy/org/gradle/java/JavaApplicationOutgoingVariantsIntegrationTest.groovy

```diff
@@ -79,109 +79,61 @@ project(':consumer') {
 
```

Comment:
There might be coverage elsewhere for this, but I think this was testing something legitimate.

If you try to resolve a configuration that's not in a project that has any "JVM ecosystem" schema or attributes setup, then you should get a certain behavior when the upstream project is using java/java-library/java-application.

With the refactoring here, we're no longer testing that scenario because the jvm-ecosystem is always available in the consumer.

Reply:
Added tests to verify the behavior when `jvm-ecosystem` is not applied and when no attributes are requested.